### ISSUE TITLE
cnvEQTL: test for linearity / correlation

### ISSUE THREAD

**Issue by lgeistlinger** [waldronlab/CNVRanger#23]
@DarioS moving over from email to issue tracker:

If there are more than two copy number levels for a gene, CNVRanger does an ANOVA-like test, which is a test for differences in means. Is there a way to treat the copy number as a numeric measurement and test for linearity of response? I have copy number estimates from whole genome sequencing and a gene may have copy numbers such as two, four and nine in different samples and I want to know if increasing copy number is associated with increased abundance, rather than just different abundance. I am not interested in analysing distant copy numbers, but the copy number of the gene itself.

**Comment by lgeistlinger** [comment]
I think you mean that you would like to test for correlation between copy number and expression as eg [here](https://pubmed.ncbi.nlm.nih.gov/21862627/) or [here](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-191).

That is currently not supported but is a nice feature request, which I'll add. 

CNVRanger carries out CNV-centric EQTL testing. Given that you are interested in local (= gene dosage) effects, the approach would be to use eg `GenomicRanges::subsetByOverlaps` to subset to CNVs that directly overlap with genes, and then use `CNVRanger::cnvEQTL` with argument `window = 0`.

**Comment by DarioS** [comment]
A challenge will be that some samples will me mostly diploid, others are mostly triploid and others are mostly tetraploid. The standard median RNA-seq scaling will scale each sample to be equal for the most common copy number state to the reference sample (median expression gene might be copy number 2 in sample X but copy number 4 in sample Y). Perhaps the comparison needs to be between scaled RNA-seq values and relative gene copy numbers (each gene's copy number -  the sample's ploidy).

**Comment by lgeistlinger** [comment]
That sounds to me like needing a purity/ploidy-aware copy number caller such as [PureCN](http://bioconductor.org/packages/release/bioc/html/PureCN.html) to produce absolute integer copy numbers that could be correlated with gene expression across samples.

**Comment by DarioS** [comment]
Yes, I use [PURPLE](https://github.com/hartwigmedical/hmftools/tree/master/purity-ploidy-estimator) (not yet published) for absolute gene copy number and sample ploidy estimation. I'm hesitant to use CPM because it is relative to all other genes in the sample and can change merely because of other genes in the same sample changing.
